parser/parser.y

```diff
@@ -4591,14 +4605,53 @@ SelectLockOpt:
 
 // See https://dev.mysql.com/doc/refman/5.7/en/union.html
 UnionStmt:
-	UnionClauseList "UNION" UnionOpt SelectStmt
+	UnionClauseList "UNION" UnionOpt SelectStmtPart1 OrderByOptional SelectStmtLimit SelectLockOpt
```

Comment:
Shall we extract a common `SelectStmtPart` which is `SelectStmtPart1 |  SelectStmtPart2 | SelectStmtPart3` ?
Then we can make UnionStmt more simpler here

Reply:
`SelectStmtPart1`, `SelectStmtPart2` and `SelectStmtPart3`  are parts of `selectStmt`.  combining them into a new rule `SelectStmtPart` may cause reduce/reduce conflicts on rule `SelectStmtPart` and `SelectStmt`, I think.  Besides, `SelectStmtPart2` is significant different from the other two and can't be combined.